### PR TITLE
[IT-283] setup AWS ECS infra for Agora app

### PR DESCRIPTION
**Pull request by zaro0508**
Deploy AWS ECS infrastructure with a sample app as
a starting point for setting up Agora to run on ECS.

This template came directly from AWS docs[1]

[1] https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-ecs.html#quickref-ecs-example-1.yaml

### CODE REVIEW

**Comment by kdaily** [comment]
Let's try it! Find some time on my calendar and we can discuss.

File: templates/agora.yaml [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+      FromPort: '22'
+      ToPort: '22'
+      CidrIp: 0.0.0.0/0
+  EcsSecurityGroupALBports:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: '31000'
+      ToPort: '61000'
```
Comment (brucehoff) [inline templates/agora.yaml]: just curious:  What is the purpose of this port?

File: templates/agora.yaml [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+          Options:
+            awslogs-group: !Ref 'CloudwatchLogsGroup'
+            awslogs-region: !Ref 'AWS::Region'
+            awslogs-stream-prefix: ecs-demo-app
+        MountPoints:
+        - ContainerPath: /usr/local/apache2/htdocs
+          SourceVolume: my-vol
+        PortMappings:
+        - ContainerPort: 80
+      - Name: busybox
```
Comment (brucehoff) [inline templates/agora.yaml]: I assume this is a placeholder and we will replace 'busybox' with the name of the agora image, correct?

File: templates/agora.yaml (hunk 1 above)
Comment (zaro0508) [reply templates/agora.yaml]: AWS ECS dynamic port mapping feature assigns host ports.  The ALB forwards requests to these dynamic host ports.

File: templates/agora.yaml (hunk 2 above)
Comment (zaro0508) [reply templates/agora.yaml]: yes

**Comment by zaro0508** [comment]
It doesn't currently, but it can. just need to add the SSL cert to the ALB.

**Pull request merged**